providers/class.two-factor-email.php

```diff
@@ -74,9 +74,9 @@ public function generate_token( $user_id ) {
 	public function validate_token( $user_id, $token ) {
 		$hashed_token = get_user_meta( $user_id, self::TOKEN_META_KEY, true );
 		if ( wp_hash( $token ) !== $hashed_token ) {
-			$this->delete_token( $user_id );
 			return false;
 		}
+		$this->delete_token( $user_id );
```

Comment:
@joshbetz This will also ensure that a token can't be re-used once entered correctly. Generating fresh tokens on every 2FA login page took care of this before your other changes.